lib/beacon/admin/media_library/file_metadata.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Beacon.Admin.MediaLibrary.FileMetadata do
+  @moduledoc """
+  Metadata passed to page rendering lifecycle.
+  """
+
+  defstruct [:site, :config, :allowed_types, :path, :name, :mime_type, :size]
+
+  @type t :: %__MODULE__{
+          site: Beacon.Types.Site.t(),
+          config: list(),
+          allowed_types: list(),
+          path: String.t() | nil,
+          name: String.t() | nil,
+          mime_type: String.t() | nil,
+          size: integer() | nil
+        }
+
+  def new(site, path, name, mime_type, size) do
+    config = Beacon.Config.fetch!(site)
+
+    asset_config =
+      config
+      |> Beacon.Config.config_for_mime_type(mime_type)
+      |> Enum.into(%{})
+
+    %__MODULE__{
```

Comment:
I see some inconsistency on the "type" key:
- config.asset_formats
- metadata.allowed_types
- metadata.mime_type
- asset.file_type

The term "MIME type" was updated to "Media Type" as noted at https://www.iana.org/assignments/media-types/media-types.xhtml so we have 3 acceptable suggestions:
- `media_type` - technically the correct name
- `mime_type` - still popular
- `type`

I'm more inclined to suggest calling it just `type` because `image/*` isn't a registered and valid media type and IANA calls the main category`image` as "type", see https://www.iana.org/assignments/media-types/image/webp - that would look like:
- config.asset_types
- metadata.allowed_types
- metadata.type
- asset.type

Wdyt?

Reply:
You are correct on multiple accounts here.
- `*` is not a valid mime sub type
- naming variance.

`image/*` is  a valid `content type`, however.

I'm comfortable with your name changes. Does `content_type` hold anything for you? It can point the way on what is expected there.

Reply:
> image/* is a valid content type, however

that's a good point, I'm okay with using `content_type`

Reply:
I was also glomming onto what Plug.Conn was doing:

https://hexdocs.pm/plug/Plug.Conn.Utils.html#media_type/1

Reply:
Sorry for the confusion but giving a second thought, I think it's better to call it `media_type` for the following reasons:
- `type` is too generic
- `media_type` is technically correct
- `content_type` serves to another purpose
- having `image/*` as the media type is fine since content type and Plug accepts it 

Sounds good?

Reply:
works for me!